cmd/argocd/commands/app.go

```diff
@@ -750,7 +781,37 @@ func NewApplicationUnsetCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
 			errors.CheckError(err)
 
 			updated := false
+			if app.Spec.Source.Kustomize != nil {
+				visited := 0
```

Comment:
The `visited` variable is used in `setAppSpecOptions` to find out if anything has changed in the app. 

In the `NewApplicationUnsetCommand` we have `updated` flag. Please use it instead. E.g.:

```
c.Flags().Visit(func(f *pflag.Flag) {
    switch f.Name {
    case "nameprefix":
        updated = true
        app.Spec.Source.Kustomize.NamePrefix = ""
    case "namesuffix":
        updated = true
        app.Spec.Source.Kustomize.NameSuffix = ""
    case "kustomize-image":
        for _, kustomizeImage := range kustomizeImages {
            for i, item := range app.Spec.Source.Kustomize.Images {
                if item.Match(argoappv1.KustomizeImage(kustomizeImage)) {
                    updated = true
                    
```

Reply:
Fixed.